src/coreclr/.nuget/Microsoft.CrossOsDiag.Private.CoreCLR/Microsoft.CrossOsDiag.Private.CoreCLR.proj

```diff
@@ -4,21 +4,74 @@
     <SupportedRids>linux-x64;linux-musl-x64;linux-arm64;linux-musl-arm64;linux-arm;linux-musl-arm</SupportedRids>
   </PropertyGroup>
 
-  <Target Name="VerifyCrossDacProperties" BeforeTargets="Build;Pack">
-    <Error Condition="'$(CrossDacArtifactsDir)' == '' or !Exists('$(CrossDacArtifactsDir)')"
-          Text="Packaging the Cross OS DAC requires property CrossDacArtifactsDir to a path containing all cross OS DAC assets needed." />
+  <!--
+    In dotnet/runtime's official build, we'll extract the runtime packs to a specific directory and pass that here.
+    In the VMR, we'll restore the runtime packs as NuGet packages.
+  -->
+  <PropertyGroup Condition="'$(DotNetBuildOrchestrator)' == 'true'">
+    <CrossRuntimeExtractionRoot>$(NuGetPackageRoot)</CrossRuntimeExtractionRoot>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <SupportedRid Include="linux-x64" TargetOS="linux" TargetArchitecture="x64" HostArch="x64" CrossDacOutputDir="linux.x64.$(Configuration)/x64" />
+    <SupportedRid Include="linux-musl-x64" TargetOS="alpine" TargetArchitecture="x64" HostArch="x64" CrossDacOutputDir="linux_musl.x64.$(Configuration)/x64" />
+    <SupportedRid Include="linux-arm64" TargetOS="linux" TargetArchitecture="arm64" HostArch="x64" CrossDacOutputDir="linux.arm64.$(Configuration)/x64" />
+    <SupportedRid Include="linux-musl-arm64" TargetOS="alpine" TargetArchitecture="arm64" HostArch="x64" CrossDacOutputDir="linux_musl.arm64.$(Configuration)/x64" />
+    <SupportedRid Include="linux-arm" TargetOS="linux" TargetArchitecture="arm" HostArch="x86" CrossDacOutputDir="linux.arm.$(Configuration)/x86" />
+    <SupportedRid Include="linux-musl-arm" TargetOS="alpine" TargetArchitecture="arm" HostArch="x86" CrossDacOutputDir="linux_musl.arm.$(Configuration)/x86" />
+
+    <SupportedRid Update="@(SupportedRid)" Rid="%(Identity)" />
+
+    <TargetRuntimePacks Include="@(SupportedRid->'Microsoft.NETCore.App.Runtime.%(Rid)')" />
+
+    <PackageDownload Condition="'$(DotNetBuildOrchestrator)' == 'true'" Include="@(TargetRuntimePacks)" Version="[$(Version)]" />
+  </ItemGroup>
+
+  <Target Name="BuildCrossOSDacs">
+    <ItemGroup>
```

Comment:
Is it better to hardcode the params here instead of doing a build.proj one with the linuxdac subset? (genuine question, I know it might evaluate things, and I don't know if it breaks anything in the multiple passes of the VMR)

Reply:
That might work and is a decent idea. I'll give it a shot locally.

Reply:
Looks like this will work for the dacs, but I'll keep the separate restore step for `runtime-prereqs`.